mzLib/Test/FileReadingTests/TestPsmFromTsv.cs

```diff
@@ -76,7 +91,104 @@ public static void ReadOGlycoPsmsLocalizedGlycans()
+            peaksFile.LoadResults();
+
+            var ProteinGroupsFile = new SpectrumMatchFromTsvFile(quantifiedProteinGroupsFilePath);
+            ProteinGroupsFile.LoadResults();
+
+            var occupancyDict = new Dictionary<String, Dictionary<int, Dictionary<string, double>>>();
+            var proteinSeqRangesSeen = new Dictionary<String, List<Tuple<int, int, double>>>();
+            var proteinsNotInDb = new List<string>();
+
+            // go through the quantified peaks 
+            foreach (var peak in peaksFile)
+            {
+                // if ambiguous peptides for a peak, go through each peptide
+                var peptidesFull = peak.FullSequence.Split('|');
+                List<string> peptidesBase = new List<string>();
+                foreach (string peptide in peptidesFull) 
+                    peptidesBase.Add(IBioPolymerWithSetMods.GetBaseSequenceFromFullSequence(peptide)); 
+
+                for (int i = 0; i < peptidesFull.Length; i++)
+                {
+                    var peptideMods = Omics.SpectrumMatch.SpectrumMatchFromTsv.ParseModifications(peptidesFull[i]);
+
+                    // go through each protein of the protein group
+                    foreach (string protein in peak.ProteinGroup.SplitAndTrim(new char[] { '|', ';' }))
+                    {
+                        int offset = 0;
+                        // try setting offset to the peptide start residue index, and catch when the protein is not in the database
+                        try
+                        {
+                            offset = proteinDbDict[protein].BaseSequence.IndexOf(peptidesBase[i]);
+
+                            // if the peptide is not a substring of the protein sequence, move on to the next protein
+                            if (offset < 0)
+                                break;
+                        }
+                        catch
+                        {
+                            if (!proteinsNotInDb.Contains(protein))
+                                proteinsNotInDb.Add(protein);
+                        }
 
+                        // add the protein to the occupancy dictionary if it has not been seen, yet
+                        if (!occupancyDict.ContainsKey(protein))
+                        {
+                            occupancyDict.Add(protein, new Dictionary<int, Dictionary<string, double>>());
+                            proteinSeqRangesSeen.Add(protein, new List<Tuple<int, int, double>>());
+                        }
+
+                        proteinSeqRangesSeen[protein].Add(new Tuple<int, int, double>(offset, offset + peptidesBase[i].Length, peak.PeakIntensity));
+
+                        // get the localized modifications from the peptide full sequence and add any amino acid/modification combination not
+                        // seen yet to the occupancy dictionary
+                        foreach (KeyValuePair<int, List<string>> aaWithModList in peptideMods)
```

Comment:
In situations like this, you can use "var aaWithModList" instead of specifying the actual class